index.js

```diff
@@ -2,16 +2,11 @@ import { NativeEventEmitter, NativeModules } from "react-native";
 
 const AdyenDropIn = NativeModules.AdyenDropInPayment;
 const EventEmitter = new NativeEventEmitter(AdyenDropIn);
-const eventMap = {};
 let onPaymentProvideListener;
 let onPaymentFailListener;
 let onPaymentSubmitListener;
 let onPaymentCancelListener;
 const addListener = (key, listener) => {
-  if (eventMap[key]) {
-    return eventMap[key];
-  }
-  eventMap[key] = listener;
   EventEmitter.addListener(key, listener);
```

Comment:
we're not returning the listener here, hence `.remove()` can't work on it.

Reply:
@YouriT not sure what you mean. I'm just creating the listener here, and afterwards the listener itself will have a `remove` function which we can use to remove it, as I'm doing at `removeListeners`.

Reply:
```js
const addListener = (key, listener) => {
  EventEmitter.addListener(key, listener);
};
```
Does not return anything, hence calls like: `onPaymentProvideListener = addListener(...)` keep the variable in an `undefined` state.